src/main/java/io/debezium/connector/yugabytedb/YugabyteDBTablePoller.java

```diff
@@ -0,0 +1,138 @@
+package io.debezium.connector.yugabytedb;
+
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.kafka.connect.connector.ConnectorContext;
+import org.apache.kafka.connect.errors.ConnectException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.yb.client.GetDBStreamInfoResponse;
+import org.yb.client.YBClient;
+import org.yb.master.MasterReplicationOuterClass.GetCDCDBStreamInfoResponsePB.TableInfo;
+
+/**
+ * Poller thread which extends {@link Thread} to keep polling for new tables on YugabyteDB
+ * server side to the provided stream ID.
+ * 
+ * @author Vaibhav Kushwaha (vkushwaha@yugabyte.com)
+ */
+public class YugabyteDBTablePoller extends Thread {
+  private static final Logger LOGGER = LoggerFactory.getLogger(YugabyteDBTablePoller.class);
+  private final short MAX_RETRY_COUNT = 5;
+
+  private final YugabyteDBConnectorConfig connectorConfig;
+  private final ConnectorContext connectorContext;
+  private final YBClient ybClient;
+  private final CountDownLatch shutdownLatch;
+  private final long pollMs;
+
+  private short retryCount = 0;
+  
+  private List<TableInfo> cachedTableInfoList = null;
```

Comment:
Is Set a better option? In kafka-connect-jdbc, they are figuring out duplicates etc which is not here.

Reply:
Yes, set seems to be a better option. Changing the list to a set. Some future work is planned to also verify the table names i.e. if a new table is added then we should also verify if it is a part of the include list.

Currently, we are slowed down by a limitation on the server side that the pgschema name is only available in a certain ListTablesResponse object, to use the same in this piece of code would make the code a little bit too complicated. I was thinking of implementing those things in further releases.